XExtensionItem/XExtensionItem.m

```diff
@@ -193,7 +193,12 @@ - (id)activityViewController:(UIActivityViewController *)activityViewController
         item.attachments = ({
             id activityItem = [self activityItemForActivityType:activityType];
             
-            NSItemProvider *mainAttachment = [[NSItemProvider alloc] initWithItem:activityItem typeIdentifier:self.typeIdentifier];
+            NSString *typeIdentifier = self.typeIdentifier;
```

Comment:
I know my block-scoping weirds you out but I think this is easier to grok than the ternary:

```objc
NSString *typeIdentifier = ^NSString *{
    if (![[activityItem class] isSubclassOfClass:[self.placeholderItem class]]) {
        NSString *derivedTypeIdentifier = typeIdentifierForActivityItem(activityItem);

        if (derivedTypeIdentifier) {
            return derivedTypeIdentifier;
        }
    }

    return self.typeIdentifier;
}();
```